vault/data_source_policy_document.go

```diff
@@ -165,38 +196,54 @@ func policyDocumentDataSourceRead(d *schema.ResourceData, meta interface{}) erro
 
 		for i, ruleI := range rawRuleIntfs {
 			rawRule := ruleI.(map[string]interface{})
-			rule := &PolicyRule{
-				Path:           rawRule["path"].(string),
-				Description:    rawRule["description"].(string),
-				MinWrappingTTL: rawRule["min_wrapping_ttl"].(string),
-				MaxWrappingTTL: rawRule["max_wrapping_ttl"].(string),
+			rule := &PolicyRule{}
+			ruleFields := map[string]*string{
+				consts.FieldPath:   &rule.Path,
+				"description":      &rule.Description,
+				"min_wrapping_ttl": &rule.MinWrappingTTL,
+				"max_wrapping_ttl": &rule.MaxWrappingTTL,
+			}
+			for k, v := range ruleFields {
+				if value, ok := rawRule[k].(string); ok {
```

Comment:
Shouldn't we return error, if `!ok` ?

Reply:
Ah, yes for the required param (`path`), we should. Will update

Reply:
done here: https://github.com/hashicorp/terraform-provider-vault/pull/2445/commits/a0692781d466613cf9c4027231dde5e6096762a2